discovery/spec.md

```diff
@@ -55,305 +45,194 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+OPTIONAL, `*` means the preceding property MAY appear zero or more times, and
+`+` means the preceding property MUST appear at least once.
 
-Note: some of the terms defined below are taken from the [CloudEvents](../cloudevents/spec.md)
-specification, and are marked with a reference to the original definition. Any
-difference between the definitions is accidental and the CloudEvents version
-takes precedence.
+### Terminology
 
 This specification defines the following terms:
 
-#### Discovery Endpoint
+#### Discovery Service
 
 A compliant implementation of this specification that advertises the set of
-Services, Event Types and other metadata to aid in the creation of an Event
-Subscription.
-
-#### Service
-
-A "service" represents the entity which manages one or more event
-[sources](#source-ce) and is associated with [producers](#producer-ce) that are
-responsible for the generation of events.
-
-For example, an Object Store service might have a set of event sources where
-each event source maps to a bucket.
-
-#### Source [[CE](../cloudevents/spec.md#source)]
+Endpoints, Groups, Definitions and other metadata to aid in the programmic
+interactions with the Endpoints.
 
-The "source" is the context in which the occurrence happened. In a distributed
-system it might consist of multiple Producers. If a source is not aware of
-CloudEvents, an external producer creates the CloudEvent on behalf of the
-source.
+### Conventions
 
-#### Producer [[CE](../cloudevents/spec.md#producer)]
+#### References
 
-The "producer" is a specific instance, process or device that creates the data
-structure describing the CloudEvent.
+Discuss uri vs self properties....
 
-#### Intermediary [[CE](../cloudevents/spec.md#intermediary)]
-
-An "intermediary" receives a message containing an event for the purpose of
-forwarding it to the next receiver, which might be another intermediary or a
-[Consumer](#consumer-ce). A typical task for an intermediary is to route the
-event to receivers based on the information in the event
-[Context](../cloudevents/spec.md#context).
-
-#### Consumer [[CE](../cloudevents/spec.md#consumer)]
-
-A "consumer" receives the event and acts upon it. It uses the context and data
-to execute some logic, which might lead to the occurrence of new events.
-
-#### Subscription
-
-The request for events from a Service.
 
 ## Resource Model
 
-This section defines the resource model of a Discovery Endpoint.
-
-### Services
+This section defines the resource model of a Discovery Service.
 
-At the core of the data model is the concept of a [Service](#service). The API
-then exposes multiple ways to query over a list of Services. To help explain the
-Service resource, the following non-normative pseudo json shows its basic
-structure:
+### Endpoint
 
-(`*` means zero or more, `+` means one or more, `?` means optional)
-
-Service:
+An Endpoint is ...
 
+The following pseudo JSON shows the basic defintion of an Endpoint:
```

Comment:
Let's chat about this on the call today because I'm starting to wonder about how things like epoch and the timestamps will work in environments where the metadata isn't stored in a DB but instead is just a visualization of resources in other systems. In cases like that, it's not clear to me how easy it would be for there to be an "epoch" value without forcing some kind of DB to be created just to persist it or how we'd know when it needs to be updated based on changes in those other systems. Maybe it's ok to force that model on people, but maybe not...